### ISSUE TITLE
IllegalStateException: Something went horribly wrong while renaming. 

### ISSUE THREAD

**Issue by pirocks** [intellij-dlanguage/intellij-dlanguage#388]
https://sentry.io/francis-nixon/intellij-dlang-main/issues/459601212/

```
IllegalStateException: Something went horribly wrong while renaming. Target nane:region.d ProjectProject '/home/<removed>/rpg_workspace/<removed>' <removed> e:null
    at io.github.intellij.dlanguage.psi.impl.named.DlangIdentifierImpl.setName(DlangIdentifierImpl.java:93)
    at io.github.intellij.dlanguage.psi.references.DReference.handleElementRename(DReference.kt:268)
    at com.intellij.refactoring.rename.RenameUtil.rename(RenameUtil.java:267)
    at com.intellij.refactoring.rename.RenameUtil.doRenameGenericNamedElement(RenameUtil.java:224)
    at com.intellij.refactoring.rename.RenamePsiElementProcessor.renameElement(RenamePsiElementProcessor.java:58)
...
(132 additional frame(s) were not displayed)

Something went horribly wrong while renaming. Target nane:region.d ProjectProject '/home/<removed>/rpg_workspace/<removed>' <removed> e:null
```

**Comment by SingingBush** [comment]
recent Sentry report for similar occurring during _CompileDirty_ action:

```
java.lang.IllegalStateException: Something went horribly wrong while renaming. Target nane:Test1.d ProjectProject (name=Test1, containerState=ACTIVE, componentStore=/home/**********/Sviluppo/Progetti/DLang/Jetbrains/Test1)  e:null
	at io.github.intellij.dlanguage.psi.impl.named.DlangIdentifierImpl.setName(DlangIdentifierImpl.java:82)
	at io.github.intellij.dlanguage.psi.references.DRefe...
```

**Comment by SingingBush** [comment]
This is still a problem in 1.28.0

```
java.lang.IllegalStateException: Something went horribly wrong while renaming. Target nane:afterMakeStaticImmutable.d ProjectProject(name=DLanguage, containerState=COMPONENT_CREATED, componentStore=/home/samael/dev/projects/intellij-dlanguage) e:null
    at io.github.intellij.dlanguage.psi.impl.named.DlangIdentifierImpl.setName(DlangIdentifierImpl.java:82)
    at io.github.intellij.dlanguage.psi.references.DReference.handleElementRename(DReference.kt:268)
    at com.intellij.refactoring.rename.RenameUtilBase.rename(RenameUtilBase.java:94)
    at com.intellij.refactoring.rename.RenameUtilBase.doRenameGenericNamedElement(RenameUtilBase.java:46)
    at com.intellij.refactoring.rename.RenameUtil.doRenameGenericNamedElement(RenameUtil.java:271)
    at com.intellij.refactoring.rename.RenamePsiElementProcessorBase.renameElement(RenamePsiElementProcessorBase.java:57)
    at com.intellij.refactoring.rename.RenameProcessor.doPerformRefactoring(RenameProcessor.java:426)
    at com.intellij.refactoring.rename.RenameProcessor.performRefactoring(RenameProcessor.java:370)
    at com.intellij.refactoring.BaseRefactoringProcessor.lambda$doRefactoring$10(BaseRefactoringProcessor.java:520)
    at com.intellij.refactoring.BaseRefactoringProcessor.callPerformRefactoring(BaseRefactoringProcessor.java:573)
    at com.intellij.refactoring.BaseRefactoringProcessor.lambda$doRefactoring$11(BaseRefactoringProcessor.java:519)
    at com.intellij.openapi.application.impl.ApplicationImpl.lambda$runEdtProgressWriteAction$9(ApplicationImpl.java:979)
    at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:189)
    at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$12(CoreProgressManager.java:608)
    at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:683)
    at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:639)
    at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:607)
    at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60)
    at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:176)
    at com.intellij.openapi.progress.util.PotemkinProgress.runInSwingThread(PotemkinProgress.java:164)
    at com.intellij.openapi.application.impl.ApplicationImpl.lambda$runEdtProgressWriteAction$10(ApplicationImpl.java:979)
    at com.intellij.openapi.application.impl.ApplicationImpl.runWriteActionWithClass(ApplicationImpl.java:1003)
    at com.intellij.openapi.application.impl.ApplicationImpl.runEdtProgressWriteAction(ApplicationImpl.java:977)
    at com.intellij.openapi.application.impl.ApplicationImpl.runWriteActionWithNonCancellableProgressInDispatchThread(ApplicationImpl.java:958)
```

fixing this should also close #596

**Comment by SingingBush** [comment]
#388, #596, and #632 should be solved now by PR #712 which is now merged. Any help testing the current state of the develop branch is appreciated. This will go out in the next release.
